In [ ]:
import numpy as np



Cargamos el dataset en un array de numpy

In [ ]:

gob=np.genfromtxt("gov_10a_exp_page_linear.csv",dtype=str,delimiter=',')

gob

array([['DATAFLOW', 'LAST UPDATE', 'freq', ..., 'TIME_PERIOD',
        'OBS_VALUE', 'OBS_FLAG'],
       ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)', '14/11/22 23:00:00', 'A',
        ..., '2012', '51.2', ''],
       ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)', '14/11/22 23:00:00', 'A',
        ..., '2013', '51.6', ''],
       ...,
       ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)', '14/11/22 23:00:00', 'A',
        ..., '2018', '39.8', ''],
       ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)', '14/11/22 23:00:00', 'A',
        ..., '2019', '40.7', ''],
       ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)', '14/11/22 23:00:00', 'A',
        ..., '2020', '45.6', '']], dtype='<U34')

Observamos sus dimensiones para poder trabajar más cómodamente.

In [ ]:
print(np.size(gob[0]))
print(np.size(gob[:,0]))

11
290


Vemos que el array gobierno 2 tiene lo que vienen siendo 290 filas y 11 columnas. La primera fila se corresponde a los nombres de las variables. En este caso, vamos a imprimir las diez primeras filas para ver qué variables son numéricas.

In [ ]:
print(gob[280:290,])

[['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)' '14/11/22 23:00:00' 'A' 'PC_GDP'
  'S13' 'TOTAL' 'TE' 'SI' '2020' '51.3' '']
 ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)' '14/11/22 23:00:00' 'A' 'PC_GDP'
  'S13' 'TOTAL' 'TE' 'SK' '2012' '41.3' '']
 ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)' '14/11/22 23:00:00' 'A' 'PC_GDP'
  'S13' 'TOTAL' 'TE' 'SK' '2013' '42.6' '']
 ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)' '14/11/22 23:00:00' 'A' 'PC_GDP'
  'S13' 'TOTAL' 'TE' 'SK' '2014' '43.4' '']
 ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)' '14/11/22 23:00:00' 'A' 'PC_GDP'
  'S13' 'TOTAL' 'TE' 'SK' '2015' '45.7' '']
 ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)' '14/11/22 23:00:00' 'A' 'PC_GDP'
  'S13' 'TOTAL' 'TE' 'SK' '2016' '42.7' '']
 ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)' '14/11/22 23:00:00' 'A' 'PC_GDP'
  'S13' 'TOTAL' 'TE' 'SK' '2017' '39.6' '']
 ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)' '14/11/22 23:00:00' 'A' 'PC_GDP'
  'S13' 'TOTAL' 'TE' 'SK' '2018' '39.8' '']
 ['ESTAT:GOV_10A_EXP$DEFAULTVIEW(1.0)' '14/11/22 23:00:00' 'A' 'PC_GDP'


Vemos que tenemos como variables numéricas útiles OBS_VALUE,ya que es el valor de cierto parámetro numérico en el momento seleccionado. Para ello, vamos a calcular el array de valores del año 2012 y por otro lado, calcularemos estadísticos generales sobre todos los valores de OBS_VALUE

In [ ]:
gobnet=gob[1:290,:]
añosvalue=gobnet[:,8:10]
añosvalue=np.append(añosvalue,values=[0,0]).reshape(29,20)
val2012=añosvalue[:,1]
print(val2012)




obsvalue=gobnet[:,9]
print(type(obsvalue[0]))
obsvalue = obsvalue.astype(np.float_)
print(np.mean(obsvalue))
print(np.var(obsvalue))
print(np.sort(obsvalue))




['51.2' '56.5' '37.8' '32.9' '40.6' '39.8' '44.2' '50.5' '46.9' '45.9'
 '48.7' '49.6' '57.3' '56.8' '46.9' '46.7' '25.3' '43.3' '57.1' '41.8'
 '38.5' '40.2' '44.7' '51.5' '41.3' '43.2' '36.0' '52.1' '41.3']
<class 'numpy.str_'>
45.27439446366782
50.89650698626693
[24.2 25.3 26.2 27.4 28.1 29.1 32.4 32.5 32.8 32.9 33.2 33.2 33.3 33.3
 33.3 33.6 34.  34.2 34.3 34.4 34.5 34.7 34.8 34.8 34.8 35.  35.2 35.4
 35.5 35.6 35.7 36.  36.  36.  36.  36.2 36.2 36.4 36.5 36.9 37.4 37.5
 37.6 37.8 37.8 37.8 38.2 38.4 38.4 38.5 38.5 38.6 38.7 38.8 38.9 39.
 39.2 39.2 39.3 39.4 39.4 39.4 39.5 39.6 39.8 39.8 40.  40.2 40.4 40.4
 40.6 40.6 40.6 40.6 40.7 40.8 41.1 41.1 41.2 41.2 41.3 41.3 41.3 41.5
 41.5 41.7 41.7 41.8 41.8 41.8 41.9 42.  42.1 42.1 42.1 42.2 42.2 42.4
 42.4 42.5 42.5 42.6 42.6 42.6 42.7 42.7 42.7 42.9 42.9 43.  43.  43.1
 43.1 43.2 43.2 43.3 43.3 43.3 43.4 43.5 43.5 43.6 43.8 43.9 44.1 44.1
 44.2 44.3 44.3 44.4 44.4 44.4 44.7 44.7 44.7 44.8 44.9 44.9 45.  45.1
 45.1 45.4 45.5 45.6 45.7 4

Vemos que la media de OBS_VALUE es de 45.27 con una varianza de 50.90, también hemos enseñado el array ordenado.